### PR TITLE
REFACTOR-#5287: remove code to test getting TypeError for Series.dropna

### LINKED ISSUE #5287: REFACTOR: remove code to test getting TypeError for Series.dropna in case of wrong kwargs
https://github.com/modin-project/modin/blob/193505fdf0c984743397ba3df56262f30aee13a8/modin/pandas/test/test_series.py#L1632

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

`CodeQL` defines this place as problematic, I see no reason not to delete it.

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5287 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/test_series.py [hunk 1]
```diff
@@ -1628,10 +1628,6 @@ def test_drop_duplicates(data, keep, inplace):
 @pytest.mark.parametrize("how", ["any", "all"], ids=["any", "all"])
 def test_dropna(data, how):
     modin_series, pandas_series = create_test_series(data)
-
-    with pytest.raises(TypeError):
-        modin_series.dropna(how=None, thresh=None)
```
[the remark this answers is not in the document]
Comment (mvashishtha) [reply modin/pandas/test/test_series.py:1633]: Shouldn't we still test that passing both `how` and `thresh` gives `TypeError`?

[the remark this answers is not in the document]
Comment (anmyachev) [reply modin/pandas/test/test_series.py:1633]: `Series.dropna` doesn't have `thresh` keyword, so we shouldn't https://pandas.pydata.org/docs/reference/api/pandas.Series.dropna.html?highlight=dropna#pandas.Series.dropna

[the remark this answers is not in the document]
Comment (mvashishtha) [reply modin/pandas/test/test_series.py:1633]: Can we remove the `thresh=2` test in line 1648-1649 too? The alert also points to that: https://github.com/modin-project/modin/security/code-scanning/350

[the remark this answers is not in the document]
Comment (anmyachev) [reply modin/pandas/test/test_series.py:1633]: Sure, I missed that

**Review by mvashishtha — APPROVED** [review]

**Comment by anmyachev** [comment]
@mvashishtha let's merge

**Pull request merged**